### PR TITLE
feat(hz): support multi http method

### PR DESCRIPTION
**Pull request by FGYFFFF**
#### What type of PR is this?

feat
#### Check the PR title.

- [x] This PR title match the format: \<type\>(optional scope): \<description\>
- [x] The description of this PR title is user-oriented and clear enough for others to understand.
- [x] Attach the PR updating the user documentation if the current PR requires user awareness at the usage level. [User docs repo](https://github.com/cloudwego/cloudwego.github.io)

#### (Optional) Translate the PR title into Chinese.
hz 支持IDL中一个接口定义多个 HTTP method

#### (Optional) More detailed description for this PR(en: English/zh: Chinese).

en: hz 支持IDL中一个接口定义多个 HTTP method
zh(optional): hz supports multiple HTTP method definitions for a single interface in IDL

### CODE REVIEW

**Comment by welkeyever** [comment]
image
If this is check, it is required to **PASTE** the docs link to this pr.

File: cmd/hz/generator/handler.go [hunk 1]
```diff
@@ -44,6 +44,7 @@ type HttpMethod struct {
 	RefPackage         string // handler import dir
 	RefPackageAlias    string // handler import alias
 	ModelPackage       map[string]string
+	GenHandler         bool // Whether to generate a handler, when an idl interface corresponds to multiple http method
```
Comment (welkeyever) [inline cmd/hz/generator/handler.go:47]: If here is to emphasize the generated handler size, better use "one" instead of "a".

Comment (FGYFFFF) [reply cmd/hz/generator/handler.go]: done

File: cmd/hz/generator/handler.go [hunk 2]
```diff
@@ -78,8 +79,10 @@ func (pkgGen *HttpPackageGenerator) genHandler(pkg *HttpPackage, handlerDir, han
 					return fmt.Errorf("generate handler %s failed, err: %v", handler.FilePath, err.Error())
 				}
 
-				if err := pkgGen.updateHandler(handler, handlerTplName, handler.FilePath, false); err != nil {
-					return fmt.Errorf("generate handler %s failed, err: %v", handler.FilePath, err.Error())
+				if m.GenHandler {
```
Comment (welkeyever) [inline cmd/hz/generator/handler.go:82]: 这个如果为false，就一个handler都不生成了吗？按照这个flag的注释，应该还是会生成一个？

Comment (FGYFFFF) [reply cmd/hz/generator/handler.go:82]: 不生成的。每一个接口可能对应多个 http 路由，但是这些路由只有一个 handler。在前面解析的时候，只有一个路由的的 m.genHandler 为 true，避免生成重复的handler

**Review by welkeyever — APPROVED** [review]

**Review by li-jin-gou — APPROVED** [review]

**Pull request merged**